contracts/characters.sol

```diff
@@ -240,6 +244,46 @@ contract Characters is Initializable, ERC721Upgradeable, AccessControlUpgradeabl
         return tokenID;
     }
 
+    function burnIntoCharacter(uint256 burnId, uint256 targetCharId) external restricted {
+        uint256 burnPower = bonusPower[burnId].add(getPowerAtLevel(tokens[burnId].level));
+        require(uint(4).mul(getPowerAtLevel(tokens[targetCharId].level)) >= getTotalPower(targetCharId).add(burnPower), "Power limit");
+        address burnOwner = ownerOf(burnId);
+        if(burnOwner == address(garrison)) {
+            burnOwner = garrison.characterOwner(burnId);
+            garrison.updateOnBurn(burnOwner, burnId);
+        }
+        _burn(burnId);
+        bonusPower[targetCharId] = burnPower.add(bonusPower[targetCharId]);
+
+        emit Burned(
+            burnOwner,
+            burnId
+        );
+    }
+
+    function burnIntoSoul(uint256 burnId, address soulRecipient) external restricted {
+        uint256 soulAmount = uint256(getPowerAtLevel(tokens[burnId].level)).add(bonusPower[burnId]).div(10);
+        address burnOwner = ownerOf(burnId);
+        if(burnOwner == address(garrison)) {
+            garrison.updateOnBurn(burnOwner, burnId);
+        }
+        _burn(burnId);
+        soulSupply[soulRecipient] = soulSupply[soulRecipient].add(soulAmount);
+
+        emit Burned(
+            soulRecipient,
+            burnId
+        );
+    }
+
+    function upgradeWithSoul(uint256 targetCharId, uint256 soulAmount) public {
```

Comment:
should probably be an ownerOf check done for this action

Reply:
There's no way in the front-end to upgrade characters one doesn't own, but might be useful for some nerds to have that option available in the contract, why not?